### PR TITLE
Add capabilities to run static analysis tools

### PR DESCRIPTION
**Pull request by rshariffdeen**

### CODE REVIEW

**Review by Marti2203 — APPROVED** [review]

File: app/drivers/tools/repair/c/GenProg.py [hunk 1]
```diff
@@ -73,19 +73,19 @@ def repair(self, bug_info, config_info):
...
-            emitter.success("\t\t\t[success] {0} ended successfully".format(self.name))
+            emitter.success("\t\t\t(success) {0} ended successfully".format(self.name))
         emitter.highlight("\t\t\tlog file: {0}".format(self.log_output_path))
         self.timestamp_log_end()
 
     def save_artifacts(self, dir_info):
         emitter.normal("\t\t\t saving artifacts of " + self.name)
         dir_results = dir_info["result"]
-        dir_patch = join(self.dir_expr, "src", "repair")
+        dir_patch = join(self.dir_expr, "src", "..")
```
[the remark this answers is not in the document]
Comment (rshariffdeen) [reply app/drivers/tools/repair/c/GenProg.py:88]: yeah it should not be changed

File: app/drivers/tools/analyze/multi/Infer.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+class Infer(AbstractAnalyzeTool):
+    relative_binary_path = None
+    bug_conversion_table = {
+        "Memory Leak": "MEMORY_LEAK",
+        "Use After Free": "USE_AFTER_FREE",
+        "Double Free": "DOUBLE_FREE",
+    }
+
+    def __init__(self):
+        self.name = os.path.basename(__file__)[:-3].lower()
```
[the remark this answers is not in the document]
Comment (rshariffdeen) [reply app/drivers/tools/analyze/multi/Infer.py:23]: not at the moment

File: app/drivers/tools/analyze/c/SAVER.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+from app.core import values
+from app.core.utilities import error_exit
+from app.drivers.tools.analyze.AbstractAnalyzeTool import AbstractAnalyzeTool
+
+
+class SAVER(AbstractAnalyzeTool):
+    relative_binary_path = None
+
+    def __init__(self):
+        self.name = os.path.basename(__file__)[:-3].lower()
```
[the remark this answers is not in the document]
Comment (rshariffdeen) [reply app/drivers/tools/analyze/c/SAVER.py:17]: not at the moment

File: app/drivers/tools/repair/c/GenProg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Marti2203) [reply app/drivers/tools/repair/c/GenProg.py:88]: Okay

**Pull request merged**